plugins/SpectrumAnalyzer/SaSpectrumView.cpp

```diff
@@ -549,28 +554,42 @@ void SaSpectrumView::drawCursor(QPainter &painter)
 		painter.drawLine(m_cursor.x(), m_displayTop, m_cursor.x(), m_displayBottom);
 		painter.drawLine(m_displayLeft, m_cursor.y(), m_displayRight, m_cursor.y());
 
-		// coordinates
+		// coordinates: background box
+		QFontMetrics fontMetrics = painter.fontMetrics();
+		unsigned int const box_left = 5;
+		unsigned int const box_top = 5;
+		unsigned int const box_margin = 3;
+		unsigned int const box_height = 2*(fontMetrics.size(Qt::TextSingleLine, "0 HzdBFS").height() + box_margin);
+		unsigned int const box_width = fontMetrics.size(Qt::TextSingleLine, "-99.9 dBFS").width() + 2*box_margin;
 		painter.setPen(QPen(m_controls->m_colorLabels.darker(), 1, Qt::SolidLine, Qt::RoundCap, Qt::BevelJoin));
-		painter.drawText(m_displayRight -60, 5, 100, 16, Qt::AlignLeft, "Cursor");
+		painter.fillRect(m_displayLeft + box_left, m_displayTop + box_top,
+						 box_width, box_height, QColor(0, 0, 0, 64));
 
+		// coordinates: text
+		painter.setPen(QPen(m_controls->m_colorLabels, 1, Qt::SolidLine, Qt::RoundCap, Qt::BevelJoin));
 		QString tmps;
+
 		// frequency
 		int xFreq = (int)m_processor->xPixelToFreq(m_cursor.x() - m_displayLeft, m_displayWidth);
 		tmps = QString(std::string(std::to_string(xFreq) + " Hz").c_str());
```

Comment:
I don't think I have any opinion on that.. :D I wrote the code in standard C++ and then only converted to `QString` because I had to (since standard library data structures are apparently not good enough for Qt and they have to reinvent the wheel for every single thing..).

Reply:
Well, at least please fix those `std::string(std::to_string(...))`, those are really ugly to read :scream: .

Reply:
In that case I would rather use the QString directly; having a 
`((std::string) + (something_else)).random_function()` and just "guessing" that the result of the outer parentheses is `std::string` does not exactly seem better to me. The explicit `std::string` cast is meant to improve readability by stating what is expected to be result of the overloaded + operator.

Reply:
OK, I'll leave it up to you